src/test/java/org/junitpioneer/jupiter/SystemPropertyExtensionTests.java

```diff
@@ -236,7 +233,36 @@ class ResettingSystemPropertyTests {
 		@Nested
 		@SetSystemProperty(key = "A", value = "newer A")
 		@TestInstance(TestInstance.Lifecycle.PER_CLASS)
-		class ResettingSystemPropertyNestedTests {
+		class ResettingSystemPropertyAfterEachNestedTests {
```

Comment:
What is the difference between this test class and `ResettingSystemPropertyAfterAllNestedTests`? (Same for `EnvironmentVariableExtensionTests.java`.)